packages/vscode-extension/src/webview/views/LaunchConfigurationView.tsx

```diff
@@ -0,0 +1,267 @@
+import "./View.css";
+import "./LaunchConfigurationView.css";
+import Label from "../components/shared/Label";
+import { useLaunchConfig } from "../providers/LaunchConfigProvider";
+import { LaunchConfigurationOptions } from "../../common/LaunchConfig";
+import Select from "../components/shared/Select";
+import { useRef } from "react";
+import { valid } from "semver";
+
+function LaunchConfigurationView() {
+  const { android, appRoot, ios, isExpo, metroConfigPath, env, update, xcodeSchemes } =
+    useLaunchConfig();
+
+  return (
+    <>
+      <Label>iOS</Label>
+      <IosConfiguration
+        scheme={ios?.scheme}
+        configuration={ios?.configuration}
+        update={update}
+        xcodeSchemes={xcodeSchemes}
+      />
+      <div className="launch-configuration-section-margin" />
+
+      <Label>Android</Label>
+      <AndroidConfiguration
+        buildType={android?.buildType}
+        productFlavor={android?.productFlavor}
+        update={update}
+      />
+
+      <div className="launch-configuration-section-margin" />
+
+      <Label>App Root</Label>
+      <AppRootConfiguration appRoot={appRoot} update={update} />
+      <div className="launch-configuration-section-margin" />
+
+      <Label>metro Config Path</Label>
+      <MetroConfigPathConfiguration metroConfigPath={metroConfigPath} update={update} />
+
+      <div className="launch-configuration-section-margin" />
+
+      <Label>is Expo</Label>
+      <IsExpoConfiguration isExpo={isExpo} update={update} />
+
+      <div className="launch-configuration-section-margin" />
+    </>
+  );
+}
+
+interface iosConfigurationProps {
+  scheme?: string;
+  configuration?: string;
+  update: <K extends keyof LaunchConfigurationOptions>(
+    key: K,
+    value: LaunchConfigurationOptions[K]
+  ) => void;
+  xcodeSchemes: string[];
+}
+
+function IosConfiguration({ scheme, configuration, update, xcodeSchemes }: iosConfigurationProps) {
+  const configurationInputRef = useRef<HTMLInputElement>(null);
+
+  const onSchemeChange = (newScheme: string) => {
+    if (newScheme === "Auto") {
+      update("ios", { scheme: undefined, configuration });
+      return;
+    }
+    update("ios", { scheme: newScheme, configuration });
+  };
+
+  const onConfigurationBlur = () => {
+    const newConfiguration = configurationInputRef.current?.value;
+    if (newConfiguration === "Auto" || !newConfiguration) {
+      update("ios", { scheme, configuration: "Auto" });
+      return;
+    }
+    update("ios", { scheme, configuration: newConfiguration });
+  };
+
+  const availableXcodeSchemes = xcodeSchemes.map((xcodeScheme) => {
+    return { value: xcodeScheme, label: xcodeScheme };
+  });
+
+  availableXcodeSchemes.push({ value: "Auto", label: "Auto" });
```

Comment:
Can we use undefined as a value and not introduce special case for "Auto"?

Reply:
unfortunately no the problem lies in our constrction of makeProxy function: 
```
export function makeProxy<T extends object>(objectName: string) {
  return new Proxy<T>({} as T, {
    get(_, methodName) {
      return (...args: any[]) => {
        const currentCallId = `${instanceToken}:${globalCallCounter++}`;
        let argsWithCallbacks = args.map((arg) => {
          if (typeof arg === "function") {
            maybeInitializeCallbackMessageListener();
            const callbackId =
              callbackToID.get(arg) || `${instanceToken}:${globalCallbackCounter++}`;
            callbackToID.set(arg, callbackId);
            idToCallback.set(callbackId, arg);
            return {
              __callbackId: callbackId,
            };
          } else {
            return arg;
          }
        });

        vscode.postMessage({
          command: "call",
          callId: currentCallId,
          object: objectName,
          method: methodName,
          args: argsWithCallbacks,
        });
        return new Promise((resolve, reject) => {
          registerCallResultPromise(currentCallId, resolve, reject);
        });
      };
    },
  });
}
```
specifically that if the last argument is undefined the `...args: any[]` length it shorter and it does not send the appropriate message.